### PR TITLE
Convert bard14 spectra

### PR DESCRIPTION
**Pull request by jolielheureux**
Converting 17 SpeX Prism spectra from bard14 into standardized fits files: https://docs.google.com/spreadsheets/d/11o5NRGA7jSbHKaTNK7SJnu_DTECjsyZ6rY3rcznYsJk/edit#gid=0

Separating header and convert to fits functions

### CODE REVIEW

**Comment by kelle** [comment]
Looks like good progress!

File: scripts/spectra_convert/convert_bard14_spectra.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        'RA' : row['ra'] ,
+        'dec' : row['dec'] ,
+        'bandpass' : row['regime'] ,
+        'aperture' : row['aperture'] ,
+        'object_name' : object_name,
+        'bibcode' : '2014ApJ...794..143B' ,
+        'instrument' : row['instrument'] ,
+        'obs_date' : row['observation date'] ,
+        'title' : None ,
+        'author' : 'Bardalez Gagliuffi, Daniella ; Burgasser, Adam ; Gelino, Christopher ; Looper, Dagny ; Nicholls, Christine ; Schmidt, Sarah ; Cruz, Kelle ; West, Andrew ; Gizis, John ; Metchev, Stanimir ',
```
Comment (kelle) [inline scripts/spectra_convert/convert_bard14_spectra.py:47]: Can just be `Bardalez Gagliuffi, et. al.`

File: scripts/spectra_convert/header_function.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+    required_keywords = ['VOCLASS', 'VOPUB', 'RA', 'dec', 'bandpass', 'aperture', 'object_name', 'bibcode', 'instrument', 'obs_date', 'title', 'author', 'doi', 'telescope', 'history', 'comment', 'observatory']
+
+    keywords_given = list(spectra_data_info.keys())
+
+    for i in range(len(keywords_given)):
+        if keywords_given[i] not in required_keywords:
+            raise Exception("Must Add Keyword")
+        else :
+            pass
```
Comment (kelle) [inline scripts/spectra_convert/header_function.py:19]:
```suggestion
    for key in keywords_given:
        if key not in required_keywords:
            raise Exception(f"Expecting keyword: {key}")
        else :
            pass
```

File: scripts/spectra_convert/identify unreadable spectra.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+# change it to a list instead of a set
+# instead of add row it is appned
+# save the list as an astropy table for the spectra -> can be txt
+# push
+# commit and push !!!!!!!!!!!!!!!!!!
+#open pull request once its in spectra convert folder
+
+from scripts.ingests.utils import *
+import csv
+from scripts.ingests.utils import *
```
Comment (kelle) [inline scripts/spectra_convert/identify unreadable spectra.py:13]: duplicated import
```suggestion
```

File: scripts/spectra_convert/convert_to_fits_function.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+def convert_to_fits(spectrum_info_all, spectrum_table, header):
+    # TODO: add docstring with expected keywords
+    # TODO: add error handling for expected keywords
+
+    object_name =header['OBJECT']
+
+ # gives original name of file
+   # spectrum_info_all['history'] = spectrum_info_all['generated_history']  # shows where file came from
+
```
Comment (kelle) [inline scripts/spectra_convert/convert_to_fits_function.py:33]: add another 'history' card here stating that the spectrum was made with the SIMPLE convert_to_fits.py function.

`header['HISTORY'] = "File made with the SIMPLE convert_to_fits.py function"`

Make sure this ADDS an additional history card and doesn't overwrite.

**Comment by jolielheureux** [comment]
[bard14.zip](https://github.com/SIMPLE-AstroDB/SIMPLE-db/files/11152231/bard14.zip)
16 converted Bard14 spectra fits files and plots of the spectra

**Comment by jolielheureux** [comment]
[bard14.zip](https://github.com/SIMPLE-AstroDB/SIMPLE-db/files/11153243/bard14.zip)
17 converted Bard14 spectra fits files and plots of the spectra

**Pull request closed without merging**